### ISSUE TITLE
Deutsche Bahn API integration

### ISSUE THREAD

**Issue by firefoxlover** [osmandapp/OsmAnd#16472]
# 🚀 feature request

### Description
When in a city it is essential to be able to use public transportation easily. Google maps is far better with this feature lacking.

Öffi by Schildbach accesses the API and also allows showing directions on a Map and finding the closest public transport station to you.

### Describe the solution you'd like

Integrate the Deutsche Bahn API as a choosable opion for the "public transport" profile.

[here is a webpage where you supposedly get APIs](https://developers.deutschebahn.com/db-api-marketplace/apis/frontpage)

### Describe alternatives you've considered
Switching between Öffi/ proprietary app and OSMAnd

**Comment by ceever** [comment]
Oppose!
I don't understand how much more OsmAnd is supposed to cover?!? It is already doing a great job! More and more features just make maintaining the app harder and harder. It also makes using the harder and harder. Why compete with apps that are already doing a great job?

MapsMe, for instance, was a great app enjoyed by many, because it was simple and useful, kind of the opposite of OsmAnd. Don't get me wrong, I like OsmAnd ... but I am also a poweruser.

OsmAnd really needs to pay closer attention to its core strength, instead of getting sidetracked by millions of additional features.

**Comment by firefoxlover** [comment]
OSMAnd has a "public transport" profile that doesnt work in germany. As far as I understood all that is nessecary is to implement the APIs and the DB does everything. Its also only one, because the DB covers whole germany and even more around.

Öffi hasnt had updates for some time often, but I am not sure if there are changes making maintenance hard.

The public transport profile just doesnt work at the moment, and Öffis navigation doesnt work. So OSMAnd is the only app that could implement this to compete with Google maps.

**Comment by scaidermern** [comment]
Of course it would be really nice to have such a feature. On the other hand, this will only be of use for a rather small percentage of users. And there are several major core functions that definitely need attention (routing speed, search engine, rendering bugs) which will benefit _all_ users.

It is fine to create such a feature request. But implementing it won't magically make OsmAnd superior to Google Maps. Especially not outside of Germany. So in my option, while being really nice to have, the importance of this feature is rather low.

Maybe it is possible to add DB API integration via an external plugin. I'm not familiar with the OsmAnd plugin API, so I can't tell if this is possible at all. If it isn't then maybe a first step is to make OsmAnd's plugin API ready for external public transport APIs.

**Comment by firefoxlover** [comment]
good idea, haven't though of that! Plugins would be a way better solution for many problems.

**Comment by firefoxlover** [comment]
I saw you now have a public transport mode, but in Germany it doesnt work at all.

https://gitlab.com/oeffi/public-transport-enabler

Oeffi, a very well working public transport app for all transports in all of Germany, has this library that could be used for routing in Germany
